test/test.js

```diff
@@ -106,6 +106,12 @@ describe('Browser', function() {
     await neverResolves;
     expect(error.message).toContain('Protocol error');
   }));
+  it('should reject if executable path is invalid', SX(async function() {
+    let waitError = null;
+    const launchOptions = Object.assign({ defaultBrowserOptions }, { executablePath: 'random-invalid-path' });
```

Comment:
you want to extend default options with executable path:
```js
const launchOptions = Object.assign({ executablePath: 'random-invalid-path' }, defaultBrowserOptions);
```

Reply:
I made a small change, but I don't think we should have `{ executablePath: ... }` as the first argument to `Object.assign(...)` because if `process.env.CHROME` is supplied, then the `executablePath` supplied in `Object.assign(...)` will be overridden causing this test to fail.

```js
const options = Object.assign({}, defaultBrowserOptions, {executablePath: 'random-invalid-path'});
```

https://github.com/GoogleChrome/puppeteer/pull/751/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R111